### PR TITLE
GRPC proto lint fixes

### PR DESCRIPTION
**Pull request by dirkkul**
### What's being changed:

- Fixes protolint errors for fields that are not used by the old client.
- Adds protolint ci job

### Review checklist

- [ ] Documentation has been updated, if necessary. Link to changed documentation:
- [ ] Chaos pipeline run or not necessary. Link to pipeline:
- [ ] All new code is covered by tests where it is reasonable.
- [ ] Performance tests have been run or not necessary.

### CODE REVIEW

File: grpc/weaviate.proto [hunk 1]
```diff
@@ -110,18 +115,19 @@ message Filters {
     bool value_bool = 6;
     float value_float = 7;
     google.protobuf.Timestamp value_date = 8;
-    strArray value_str_array = 9;
-    intArray value_int_array = 10;
-    boolArray value_bool_array = 11;
-    floatArray value_float_array = 12;
-    dateArray value_date_array = 13;
+    StrArray value_str_array = 9;
```
Comment (antas-marcin) [inline grpc/weaviate.proto:118]: Can we rename the type from `StrArray` to `StringArray`? (I think str is used only in Python to describe string) or shouldn't it be here `TextArray` as we have deprecated `String` type?

File: grpc/weaviate.proto [hunk 2]
```diff
@@ -110,18 +115,19 @@ message Filters {
...
+    IntArray value_int_array = 10;
+    BoolArray value_bool_array = 11;
```
Comment (antas-marcin) [inline grpc/weaviate.proto:120]: We also use `boolean` instead of `bool`, can we change this also?

File: grpc/weaviate.proto [hunk 3]
```diff
@@ -110,18 +115,19 @@ message Filters {
...
+    FloatArray value_float_array = 12;
```
Comment (antas-marcin) [inline grpc/weaviate.proto:121]: Instead of `Float` it's `Number`

File: grpc/weaviate.proto [hunk 4]
```diff
@@ -239,10 +247,10 @@ message ResultProperties {
   repeated ReturnRefProperties ref_props = 2;
   string class_name = 3;
   ResultAdditionalProps metadata = 4;
-  repeated FloatArrayProperties floatArrayProperties = 5;
+  repeated NumberArrayProperties numberArrayProperties = 5;
```
Comment (antas-marcin) [inline grpc/weaviate.proto:250]: why here we are not using a snake case for variable names?

Comment (dirkkul) [reply grpc/weaviate.proto]: added a wrong ignore command for the linter. Fixed now

File: grpc/weaviate.proto [hunk 5]
```diff
@@ -67,66 +70,71 @@ message SearchRequest {
...
 }
 
 
 message AdditionalProperties {
   bool uuid = 1;
   bool vector = 2;
+  // protolint:disable:next FIELD_NAMES_LOWER_SNAKE_CASE
   bool creationTimeUnix = 3;
+  // protolint:disable:next FIELD_NAMES_LOWER_SNAKE_CASE
   bool lastUpdateTimeUnix = 4;
```
Comment (antas-marcin) [inline grpc/weaviate.proto:133]: here `creationTimeUnix` and  `lastUpdateTimeUnix` we leave with camel case, yes?

File: grpc/weaviate.proto [hunk 6]
```diff
@@ -67,66 +70,71 @@ message SearchRequest {
...
   bool distance = 5;
   bool certainty = 6;
   bool score = 7;
+  // protolint:disable:next FIELD_NAMES_LOWER_SNAKE_CASE
   bool explainScore = 8;
```
Comment (antas-marcin) [inline grpc/weaviate.proto:138]: `explainScore` also camel case?

**Review by antas-marcin — APPROVED** [review]

**Pull request merged**